### PR TITLE
UCT/TAG: Add stats for tag API

### PR DESCRIPTION
**Pull request by brminich**
## What
Add statistics for TAG API in UCT

## Why ?
This type of statistics would be very useful for profiling UCX based applications

### CODE REVIEW

**Comment by shamisp** [comment]
You must document this on wiki. This is awesome statistic to have.

**Comment by brminich** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7710/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7716/ for details (Mellanox internal link).

File: src/uct/ib/rc/accel/rc_mlx5_common.h [hunk 1]
```diff
@@ -1097,6 +1097,10 @@ uct_rc_mlx5_iface_common_tag_recv(uct_rc_mlx5_iface_common_t *iface,
                                          tag_mask,
                                          UCT_RC_MLX5_SRQ_FLAG_TM_CQE_REQ |
                                          UCT_RC_MLX5_SRQ_FLAG_TM_SW_CNT);
+
+    UCT_RC_IFACE_TM_STAT(rc_iface, LIST_ADD);
+    UCT_RC_IFACE_TM_STAT(rc_iface, LIST_SYNC);
```
Comment (yosefe) [inline src/uct/ib/rc/accel/rc_mlx5_common.h]: i think it's better to count sync ops, only when they are done individually, and not as piggyback on another operation

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7752/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
Please submit News PR update for the new feature. thanks

**Comment by brminich** [comment]
@shamisp, #2899

**Pull request merged**